Minor compiler warning on build, fread return value unused

Compiling picotool under Ubuntu 20.04 and GCC 9.4 results in the following warning about not checking the the source file read completed properly:

````
Scanning dependencies of target picotool
[ 25%] Building CXX object CMakeFiles/picotool.dir/main.cpp.o
/home/earle/src/picotool/main.cpp: In member function ‘virtual void file_memory_access::read(uint32_t, uint8_t*, uint32_t, bool)’:
/home/earle/src/picotool/main.cpp:869:22: warning: ignoring return value of ‘size_t fread(void*, size_t, size_t, FILE*)’, declared with attribute warn_unused_result [-Wunused-result]
  869 |                 fread(buffer, this_size, 1, file);
      |                 ~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~
[ 50%] Building CXX object CMakeFiles/picotool.dir/picoboot_connection/picoboot_connection_cxx.cpp.o
[ 75%] Building C object CMakeFiles/picotool.dir/picoboot_connection/picoboot_connection.c.o
[100%] Linking CXX executable picotool
[100%] Built target picotool
````

--- a/main.cpp
+++ b/main.cpp
@@ -868,9 +868,7 @@ struct file_memory_access : public memory_access {
                 fseek(file, result.second + result.first.offset, SEEK_SET);
                 auto read_bytes = fread(buffer, 1, this_size, file);
                 if (this_size != read_bytes) {
-                    std::stringstream err;
-                    err << "File read error, wanted " << this_size << ", only read " << read_bytes << "bytes.";
-                    throw std::runtime_error(err.str().c_str());
+                    fail(ERROR_READ_FAILED, "Source file read error");
                 }
             } catch (not_mapped_exception &e) {
                 if (zero_fill) {

--- a/main.cpp
+++ b/main.cpp
@@ -866,7 +866,12 @@ struct file_memory_access : public memory_access {
                 this_size = std::min(size, result.first.max_offset - result.first.offset);
                 assert(this_size);
                 fseek(file, result.second + result.first.offset, SEEK_SET);
-                fread(buffer, this_size, 1, file);
+                auto read_bytes = fread(buffer, 1, this_size, file);
+                if (this_size != read_bytes) {
+                    std::stringstream err;
+                    err << "File read error, wanted " << this_size << ", only read " << read_bytes << "bytes.";
+                    throw std::runtime_error(err.str().c_str());
+                }
             } catch (not_mapped_exception &e) {
                 if (zero_fill) {
                     // address is not in a range, so fill up to next range with zeros

